client.go

```diff
@@ -273,16 +274,32 @@ func (c *Client) GetTask(taskID int64) (resp *Task, err error) {
 	return resp, nil
 }
 
-func (c *Client) GetTasks() (resp *ResultTask, err error) {
-	resp = &ResultTask{}
+func (c *Client) GetTasks(param *TasksQuery) (resp *TaskResult, err error) {
+	resp = &TaskResult{}
 	req := internalRequest{
 		endpoint:            "/tasks",
 		method:              http.MethodGet,
 		withRequest:         nil,
 		withResponse:        &resp,
+		withQueryParams:     map[string]string{},
 		acceptedStatusCodes: []int{http.StatusOK},
 		functionName:        "GetTasks",
 	}
+	if param != nil && param.Limit != 0 {
+		req.withQueryParams["limit"] = strconv.FormatInt(param.Limit, 10)
+	}
+	if param != nil && param.From != 0 {
+		req.withQueryParams["from"] = strconv.FormatInt(param.From, 10)
+	}
+	if param != nil && param.Status != "" {
+		req.withQueryParams["status"] = param.Status
+	}
+	if param != nil && param.Type != "" {
+		req.withQueryParams["type"] = param.Type
```

Comment:
I think you can define the map inline, and after that, remove the nulls or empties like this:

```go
map[string]string{
   "type": len(param.IndexUID) != 0 ? strings.Join(param.IndexUID, ",") : nil, 
   "limit": ...
}
```

Reply:
You cannot put logic inline in a struct declaration in go or I don't know how